core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -714,4 +714,18 @@ public static String generateBadRecordsPath(String badLogStoreLocation, String s
           CarbonCommonConstants.FILE_SEPARATOR + taskNo;
     }
   }
+
+  /**
+   * Return the parent path of the input file.
+   * For example, if input file path is /user/warehouse/t1/file.carbondata
+   * then return will be /user/warehouse/t1
+   */
+  public static String getParentPath(String dataFilePath) {
+    int endIndex = dataFilePath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR);
+    if (endIndex > -1) {
```

Comment:
can we replace if else with conditional operator here?

Reply:
I think this is better, especially for debugging